src/main/java/edu/cornell/kfs/coa/document/validation/impl/GlobalIndirectCostRecoveryAccountsRule.java

```diff
@@ -183,208 +169,193 @@ protected boolean checkIndirectCostRecoveryAccount(IndirectCostRecoveryAccountCh
-						}
-					}
+			}
+			
+			if (icrAccountsToBeUpdatedToActive.size() > 0) {
+				for (IndirectCostRecoveryAccount icrAccount : icrAccountsToBeUpdatedToActive) {
+					currentActiveIndirectCostRecoveryAccountList.add(icrAccount);
 				}
+			}
 
-				if ((ObjectUtils.isNull(currentActiveIndirectCostRecoveryAccountList) || (currentActiveIndirectCostRecoveryAccountList.size() == 0))) {
-					return true;
+			if (icrAccountsToBeAddedAsActive.size() > 0) {
+				for (IndirectCostRecoveryAccount icrAccount : icrAccountsToBeAddedAsActive) {
+					currentActiveIndirectCostRecoveryAccountList.add(icrAccount);
 				}
+			}
 
-				BigDecimal totalDistribution = BigDecimal.ZERO;
-
-				if (ObjectUtils.isNotNull(currentActiveIndirectCostRecoveryAccountList) && (currentActiveIndirectCostRecoveryAccountList.size() > 0)) {
-					for (IndirectCostRecoveryAccount icra : currentActiveIndirectCostRecoveryAccountList) {
-						totalDistribution = totalDistribution.add(icra.getAccountLinePercent());
+			if (ObjectUtils.isNotNull(icrAccountsNotUpdated) && !icrAccountsNotUpdated.isEmpty()) {
+				for (IndirectCostRecoveryAccount icrAccount : icrAccountsNotUpdated) {
+					if (icrAccount.isActive()) {
+						currentActiveIndirectCostRecoveryAccountList.add(icrAccount);
 					}
 				}
-
-				// check the total distribution is 100
-				if (totalDistribution.compareTo(BD100) != 0) {
-					putFieldError(CUKFSPropertyConstants.SUB_ACCOUNT_GLBL_CHANGE_DETAILS, CUKFSKeyConstants.ERROR_DOCUMENT_GLB_MAINT_ICR_ACCOUNT_TOTAL_NOT_100_PERCENT, new String[] { subAccountGlobalDetail.getChartOfAccountsCode() + "-"+ subAccountGlobalDetail.getAccountNumber() + "-" + subAccountGlobalDetail.getSubAccountNumber() });
-					result &= false;
-				}
 			}
 		}
-
-		return result;
-	}
-	
-	/**
-	 * Creates an IndirectCostRecoveryAccount from the global icr change object.
-	 * 
-	 * @param subAccountGlobalDetail
-	 * @param newICR
-	 * @return an IndirectCostRecoveryAccount
-	 */
-	private IndirectCostRecoveryAccount createIndirectCostRecoveryAccountFromChange(SubAccountGlobalDetail subAccountGlobalDetail, IndirectCostRecoveryAccountChange newICR){
-		String chart = subAccountGlobalDetail.getChartOfAccountsCode();
-		String account = subAccountGlobalDetail.getAccountNumber();
 		
-		IndirectCostRecoveryAccount icrAccount = new IndirectCostRecoveryAccount();
-		icrAccount.setAccountNumber(account);
-		icrAccount.setChartOfAccountsCode(chart);
-		icrAccount.setIndirectCostRecoveryAccountNumber(newICR.getIndirectCostRecoveryAccountNumber());
-		icrAccount.setIndirectCostRecoveryFinCoaCode(newICR.getIndirectCostRecoveryFinCoaCode());
-		icrAccount.setActive(newICR.isActive());
-		icrAccount.setAccountLinePercent(newICR.getAccountLinePercent());
+		return  currentActiveIndirectCostRecoveryAccountList;
+    }
+
+	private String buildMessageFromPrimaryKey(GlobalBusinessObjectDetailBase detail){
+		StringBuffer message = new StringBuffer();
```

Comment:
Unless you have a specific need for using StringBuffer's thread-safety or regex-related features, I would suggest using StringBuilder instead for improved efficiency.